### PR TITLE
Replace logsafe with safecli for safe CLI handling

### PR DESCRIPTION
**Pull request by plar**
## Change Overview

This PR proposes replacing the `logsafe` package with the `safecli` package for command handling and logging.

Key Differences:

1. **Command Builder Pattern**: The `Builder` structure in `safecli` offers a clearer way to assemble commands, improving code maintainability.

1. **Type Safety**: The use of specific interfaces in `safecli` for command building and logging.

1. **Formatting**: `safecli` provides distinct formatters for command arguments and logging, offering more flexibility.

1. **Design**: `safecli`’s modular design is more extensible for future enhancements.

1. **Testing**: `safecli` also includes comprehensive tests, which are currently missing in `logsafe`.

The transition to `safecli` aims to improve our project's command handling capabilities and maintainability.

## Pull request type

Please check the type of change your PR introduces:
- [ ] :construction: Work in Progress
- [x] :rainbow: Refactoring (no functional changes, no api changes)
- [ ] :hamster: Trivial/Minor
- [ ] :bug: Bugfix
- [x] :sunflower: Feature
- [ ] :world_map: Documentation
- [ ] :robot: Test

## Test Plan

- [ ] :muscle: Manual
- [x] :zap: Unit test
- [ ] :green_heart: E2E

### CODE REVIEW

File: pkg/safecli/doc.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+//     // which returns a log string representation of the command.
+//     // This is similar to calling logger.Log() as shown below.
+//
+//     // Build the command.
+//     command := zipcli.Build()
+//     fmt.Println("Command:", command)
+//     // Output: Command: [zip --temp-path=/tmp --exclude=*.log -p=password123 project_backup.zip ~/home/user/project]
+//
+//     // Log the command with sensitive data redacted.
+//     logger := safecli.NewLogger(zipcli)
```
[the remark this answers is not in the document]
Comment (plar) [reply pkg/safecli/doc.go:57]: Here are main reasons:

#### Separation of Concerns 
 
Builder makes commands. Logger is for logging. They do similar things now, but their jobs are different. This helps to keep code organized.

#### Future Changes
 
 Now, logger looks like builder's `String()` method. But in future, we might add new things to logger for better logging. Then, having separate logger helps.

#### Safe Logging

Logger makes sure sensitive data is always hidden in logs. This is very important and logger is specially made for this.

So, the functionality seems overlapping but this design provides a robust foundation for future and better code organization.

File: pkg/safecli/formatter.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+package safecli
+
+import "strings"
+
+// ArgumentFormatter defines a function that formats a command argument to the string.
+type ArgumentFormatter func(a Argument) string
+
+// format combines arguments using ArgumentFormatter.
+func (f ArgumentFormatter) format(args []Argument) []string {
```
Comment (e-sumin) [inline pkg/safecli/formatter.go:23]: I'm not saying that this is something unclear for me, but I'd prefer something more clear. Something like
```suggestion
func formatArguments(f ArgumentFormatter, args []Argument) []string {
```

File: pkg/safecli/safecli.go [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+
+// Redactor defines an interface for handling sensitive value in the way
+// that it can be represented both in a plain and redacted text form.
+type Redactor interface {
+	PlainString() string
+	String() string
+}
+
+// redactor defines a function that converts value to Redactor.
+type redactor func(value string) Redactor
```
Comment (e-sumin) [inline pkg/safecli/safecli.go:32]: I'd prefer to have more clear name for this function. Let's say `createRedactor` or something like that.

**Comment by plar** [comment]
PR is closed and a new project [github.com/kanisterio/safecli](https://github.com/kanisterio/safecli) is created.

**Comment by plar** [comment]
@e-sumin please review https://github.com/kanisterio/safecli/pull/2

**Pull request closed without merging**